### ISSUE TITLE
J2CA0045E -Tolerating the connection error occurred event during the matchManagedConnention

### ISSUE THREAD

**Issue by gjwatts** [OpenLiberty/open-liberty#14935]
**Describe the bug**
J2CA0045E: Connection not available while invoking method createOrWaitForConnection for resource

Tolerating the connection error occurred event during the matchManagedConnection. The connection pool when processing the event is losing one or more managed connections during the error event.

**Comment by tschurins** [comment]
Hello,
We have the same issue when using MQ (after a broker restart). Has this been patched?

**Comment by gjwatts** [comment]
@tschurins Hi Thomas.  Thanks for asking about this one.  No, this has not been fixed yet, but since you have reminded me, I see I was very close to finishing it.  I think I have the code changes complete now and will try to get the testing and code review completed soon to try and get this finished sometime next week.

**Comment by gjwatts** [comment]
@tschurins Hi Thomas.  Sorry, this fix is taking longer than expected.  The test to verify this is failing and I have not had time to look into it further yet.  But I will keep working on this.

**Comment by tschurins** [comment]
Hello,
Did you make any progress?
If I can help with something, just shoot...

**Comment by gjwatts** [comment]
@tschurins Sorry Thomas, no progress yet.  I don't know what to ask of you to help.  I need to better understand why the test is failing.  It gets to this line in my current PR [FreePool.java](https://github.com/OpenLiberty/open-liberty/blob/664e57f3332a493e869edcddaaab8f5eccf10119/dev/com.ibm.ws.jca.cm/src/com/ibm/ejs/j2c/FreePool.java#L695), and I am expecting `if (((com.ibm.ejs.j2c.MCWrapper) mcWrapperTemp1).errorDuringExternalCall)` to be true, but it must be false since I get this in my log:
```
[9/21/23, 12:46:59:983 CDT] 0000002d id=00000000 com.ibm.ejs.j2c.FreePool           3 GJW19Sep2023 This should show up in the logs!
[9/21/23, 12:46:59:983 CDT] 0000002d id=00000000 com.ibm.ejs.j2c.FreePool           3 GJW19Sep2023 This was not expected...
```

**Comment by gjwatts** [comment]
@tschurins I have a meeting set up the week of the 23rd to discuss this with the person who can help me with this.  I hope to get this figured out then.

**Comment by tschurins** [comment]
Hello,
How was the meeting?
Did it shed light on the way to solve the issue?

**Comment by gjwatts** [comment]
@tschurins Hi Thomas.  We met last week and had to work through a problem we were having trying to recreate the issue.  We then were able to make a code change we thought would fix the problem, but it still failed.  We have another meeting tomorrow to continue to work on this problem and figure it out.  Thanks for reaching out and letting us know this issue is important to you, that makes it important to us.

**Comment by gjwatts** [comment]
@tschurins Hi Thomas.  I think we have the problem figured out now.  I will kick off an internal verification build next week and if that goes well, I hope to have it merged by the end of next week.

**Comment by gjwatts** [comment]
@tschurins Hi Thomas.  This fix has been merged.

**Comment by tschurins** [comment]
Amazing, I'm eager to use this!

**Comment by gjwatts** [comment]
@tschurins I am glad to hear that and I hope it resolves your issue.

The fix should be available in 23.0.0.12 or you can try one of the [recent nightly builds](https://openliberty.io/start/#development_builds) that has this change.
